Sources/Confidence/ConfidenceValue.swift

```diff
@@ -0,0 +1,148 @@
+import Foundation
+
+public typealias ConfidenceStruct = [String: ConfidenceValue]
+
+/// Serializable data structure meant for event sending via Confidence
+public enum ConfidenceValue: Equatable, Encodable {
```

Comment:
I like the main gist of this but I was wondering if we could "close down" the constructing of these types?
What I'm after is to have the "enum constructors" being private and have public "create functions" where we can be enforce type safety (mainly for `list`).